checker.cs

```diff
@@ -3,36 +3,168 @@
 
 class Checker
 {
-    static bool batteryIsOk(float temperature, float soc, float chargeRate) {
-        if(temperature < 0 || temperature > 45) {
-            Console.WriteLine("Temperature is out of range!");
+   /// <summary>
+    /// Batteries the is ok.
+    /// </summary>
+    /// <param name="temperature">The temperature.</param>
+    /// <param name="soc">The soc.</param>
+    /// <param name="chargeRate">The charge rate.</param>
+    /// <returns></returns>
+    static bool batteryIsOk(float temperature, float soc, float chargeRate)
+    {
+        bool temperatureConstrainCheck = checkTemperature(temperature);
+        bool chargeStateConstrainCheck = checkStateOfCharge(soc);
+        bool chargeRateConstrainCheck = checkChargeRate(chargeRate);
+        return (temperatureConstrainCheck && chargeStateConstrainCheck && chargeRateConstrainCheck);
+    }
+
+    /// <summary>
+    /// Checks the temperature.
+    /// </summary>
+    /// <param name="temperature">The temperature.</param>
+    /// <returns></returns>
+    static bool checkTemperature(float temperature)
+    {
+        if (temperature < 0 || temperature > 45)
+        {
+            EvaluateHighTemperature(temperature);
+            EvaluateLowTemperature(temperature);
+            DisplayOutOfRangeMessage("Temperature");
             return false;
-        } else if(soc < 20 || soc > 80) {
-            Console.WriteLine("State of Charge is out of range!");
+        }
+        return true;
+    }
+
+    /// <summary>
+    /// Checks the state of charge.
+    /// </summary>
+    /// <param name="soc">The soc.</param>
+    /// <returns></returns>
+    static bool checkStateOfCharge(float soc)
+    {
+        if (soc < 20 || soc > 80)
+        {
+            EvaluateHighStateOfCharge(soc);
```

Comment:
Most of the evaluate functions have same logic ....DRY (Do not repeat your self)

Reply:
Now I have one only EvaluateBatteryMeasure, which can be used for any Battery Measure (temperature, charge rate, state of charge). And I have defined one more class which has Measure value, name, maximum and minimum limit to evaluate the battery breach and print the respective message.